sdlib/d/gc/arena.d

```diff
@@ -338,6 +389,23 @@ private:
 		unusedExtents.insert(e);
 	}
 
+	void shrinkAllocImpl(Extent* e, uint n, uint oldPages, uint newPages) {
+		assert(mutex.isHeld(), "Mutex not held!");
+		assert(n <= PageCount - oldPages, "Invalid index!");
+		assert(modUp(e.size / PageSize, PageCount) == oldPages,
+		       "Unexpected number of old pages!");
+		assert(newPages < oldPages && newPages > 0, "Invalid new page count!");
+
+		auto hpd = e.hpd;
+		unregisterHPD(hpd);
+
+		e.at(e.address, newPages * PageSize, hpd);
+		hpd.clear(n + newPages, oldPages - newPages);
```

Comment:
This tells me what you actually need as input is `e`, `n + newPages`, `newPages`, `delta`. It's 4 just as before, but it maps to what you actually need in there.

Reply:
Probably named `e`, `index`, `pages`, `delta`.